src/microbe_stage/organelle_components/MovementComponent.cs

```diff
@@ -28,6 +30,17 @@ public MovementComponent(float momentum)
 
     public void OnAttachToCell(PlacedOrganelle organelle)
     {
+        var configuration = organelle.Upgrades?.CustomUpgradeData;
+
+        if (configuration == null)
+        {
+            flagellumLength = 0;
+        }
+        else
+        {
+            flagellumLength = ((FlagellumUpgrades)configuration).LengthFraction;
```

Comment:
For safety the if here should be rewritten as `if (organelle.Upgrades?.CustomUpgradeData is FlagellumUpgrades flagellumUpgrade)`